internal/cmd/flags.go

```diff
@@ -42,7 +43,14 @@ var (
 		Usage: "signature envelope format, options: 'jws', 'cose'",
 	}
 	SetPflagSignatureFormat = func(fs *pflag.FlagSet, p *string) {
-		fs.StringVar(p, PflagEnvelopeType.Name, envelope.JWS, PflagEnvelopeType.Usage)
+		defaultEnvelopeFormat := envelope.JWS
+		// load config to get envelopeType
+		config, err := configutil.LoadConfigOnce()
+		if err == nil && config.EnvelopeType != "" {
```

Comment:
Correct me if I'm wrong, config.EnvelopeType value is from config.json, and a user can modify this value to any string right? If that's the case, do we have a logic to guarantee that config.EnvelopeType is "JWS" or "COSE"?

Reply:
Yes, if the envelopeType of config.json is an invalid value, it can be overwritten by cli flag `--envelope-type` or causes [an error ](https://github.com/notaryproject/notation/blob/main/internal/cmd/signer.go#L20). We should handle the error when we really use it.

Reply:
> Yes, if the envelopeType of config.json is an invalid value, it can be overwritten by cli flag `--envelope-type` or causes [an error ](https://github.com/notaryproject/notation/blob/main/internal/cmd/signer.go#L20). We should handle the error when we really use it.

Then [GetEnvelopeMediaType] (https://github.com/notaryproject/notation/blob/eb7e4f4696591994e3aff47d50eccdaf04f5ca3c/internal/cmd/signer.go#L58) should be a more general method. Currently, it's inside internal/cmd/signer.go, and it's only used by GetSigner. If we add a new command in the future and want to use config.json's EvelopeType, we need to call GetEnvelopeMediaType. (Btw, GetEnvelopeMediaType needs a comment/func doc.)

Reply:
Updated.